### ISSUE TITLE
BUG-When downloading a file, we don't get the suggested name, but rather the actual file name

### ISSUE THREAD

**Issue by gbritoda** [Avaiga/taipy#433]
**Description**
Could be just a simple case of the [manual on file download](https://docs.taipy.io/en/release-2.3/manuals/gui/viselements/file_download/) not being clear, but when downloading a file in the system through the `file_download` element, the `name` option doesn't seem to matter. We always get the name of the file we downloaded.

**How to reproduce**

```python
from taipy.gui import Gui

DOWNLOAD_PATH = "/home/gbritoda/repos/taipy_dev/taipy_download.py"

page = """
<|{DOWNLOAD_PATH}|file_download|name=my_downloadable_file.py|>
"""

GUI_OBJ = Gui(page=page)
GUI_OBJ.run(
        host="0.0.0.0",
        port=1234,
        dark_mode=False,
        debug=True,
        use_reloader=True,
)
```

Click download, we expect to get `my_downloadable_file.py`, but we get `taipy_download.py`, which is the original name.

**Expected behavior**
Expected to get the contents of `taipy_download.py`, but with the name `my_downloadable_file.py`.

Instead we simply get `taipy_download.py`, content and title.

**Screenshots**
![image](https://github.com/Avaiga/taipy-gui/assets/33906448/a4bbfe77-acbc-499f-8529-c9ba42b4d860)

**Runtime environment**

* OS: WSL Ubuntu 22.04.2 LTS
* Browser: Firefox 118.0.1 (64 bit)
* TaiPy 2.4.0

**Comment by FabienLelaquais** [comment]
@gbritoda This problem was indeed a bug (see [the issue](https://github.com/Avaiga/taipy-gui/issues/966) that you created the same day) that was found and fixed in Taipy GUI 3.0, that you can use right away.
Two notes though:
- the 'bypass' property of the file_download control has a default value of False, when the doc says it's True.
  So if you don't want a preview of the file (in a more general context than images or PDFs), you should force it in the control definition (`|bypass|`).
- The single incompatibility between 2.4 and 3.0 would be the 'on_action' callback function signatures. The Release Notes will tell you more and what to do.

Please let us know if that is enough for you to close that issue.

**Comment by gbritoda** [comment]
Hi @FabienLelaquais , if that's fixed in TaiPy 3.0 I am more than happy to close the issue.

Thank you for getting back 😊

**Comment by FabienLelaquais** [comment]
Thank you!
